From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Fri, 26 Feb 2021 12:08:46 +0100
Subject: [PATCH] let ninja use the jobserver

This is only for ptxdist. Ninja uses the make jobserver here, so don't add
'-j1' if no -jX argument is given.

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index bcbbec71a1cb..3868d7638e46 100644
--- a/Makefile
+++ b/Makefile
@@ -132,7 +132,7 @@ MAKE.k = $(findstring k,$(firstword $(MAKEFLAGS)))
 MAKE.q = $(findstring q,$(firstword $(MAKEFLAGS)))
 MAKE.nq = $(if $(word 2, $(MAKE.n) $(MAKE.q)),nq)
 NINJAFLAGS = $(if $V,-v) $(if $(MAKE.n), -n) $(if $(MAKE.k), -k0) \
-        $(filter-out -j, $(lastword -j1 $(filter -l% -j%, $(MAKEFLAGS)))) \
+        $(filter-out -j, $(lastword $(filter -l% -j%, $(MAKEFLAGS)))) \
 
 ninja-cmd-goals = $(or $(MAKECMDGOALS), all)
 ninja-cmd-goals += $(foreach t, $(.tests), $(.test.deps.$t))
